src/main/kotlin/com/openlattice/postgres/RetryableLockedIdsOperator.kt

```diff
@@ -0,0 +1,55 @@
+package com.openlattice.postgres
+
+import com.geekbeast.util.LinearBackoff
+import com.geekbeast.util.attempt
+import com.openlattice.data.EntityDataKey
+import com.openlattice.data.storage.lockEntitiesInIdsTable
+import org.postgresql.util.PSQLException
+import java.sql.Connection
+
+class RetryableLockedIdsOperator {
+
+    companion object {
+
+        private const val MAX_INTERVAL: Long = 60 * 1_000
+        private const val RETRY_OFFSET: Long = 125
+        private const val MAX_RETRY_ATTEMPTS = 32
+
+        fun lockIdsAndExecute(
+                connection: Connection,
+                entitySetIdToEntityKeyId: Map<EntityDataKey, Int>,
+                operateOnConnectionFn: (Connection) -> Int
+        ): Int {
```

Comment:
You can make return type generic and make it match the function that is passed in.